### PR TITLE
project refactored

### PR DESCRIPTION
**Pull request by kirosamir29**

### CODE REVIEW

File: lib/widget/todo_card.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+class TodoCard extends StatelessWidget {
```
Comment (MinaSamirCap) [inline lib/widget/todo_card.dart]: rename it to TodoCardWidget

File: lib/widget/todo_card.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  final int index;
+  final Map item;
+  final Function (Map) navigateEdit;
```
Comment (MinaSamirCap) [inline lib/widget/todo_card.dart]: rename it to navigationEditCallback

File: lib/utils/snack_helper.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+
+void showErrorMessage(BuildContext context ,{required String message}) {
```
Comment (MinaSamirCap) [inline lib/utils/snack_helper.dart]: make it one function and pass the color in it

File: lib/screens/todo_list.dart [hunk 4]
```diff
@@ -36,39 +37,25 @@ class _TodoListPageState extends State<TodoListPage> {
...
-                      ),
-                      PopupMenuItem(
-                        child: Text("Delete"),
-                        value: "delete",
-                      ),
-                    ];
-                  }),
-                );
-              }),
+          child: Visibility(
```
Comment (MinaSamirCap) [inline lib/screens/todo_list.dart]: Split in a standalone widget

File: lib/screens/todo_list.dart [hunk 5]
```diff
@@ -36,39 +37,25 @@ class _TodoListPageState extends State<TodoListPage> {
...
+            visible: items.isNotEmpty,
+            replacement: Center(
+              child: Text("No thing to do"),
+            ),
+            child: ListView.builder(
```
Comment (MinaSamirCap) [inline lib/screens/todo_list.dart]: Split in a standalone widget

**Review by MinaSamirCap — CHANGES_REQUESTED** [review]

File: lib/widget/todo_card.dart (hunk 1 above)
Comment (kirosamir29) [reply lib/widget/todo_card.dart]: done

File: lib/widget/todo_card.dart (hunk 2 above)
Comment (kirosamir29) [reply lib/widget/todo_card.dart]: done

File: lib/utils/snack_helper.dart (hunk 3 above)
Comment (kirosamir29) [reply lib/utils/snack_helper.dart]: done

File: lib/screens/todo_list.dart (hunk 4 above)
Comment (kirosamir29) [reply lib/screens/todo_list.dart]: done

File: lib/screens/todo_list.dart (hunk 5 above)
Comment (kirosamir29) [reply lib/screens/todo_list.dart]: done

**Pull request merged**